openassessment/workflow/workflow_batch_update_api.py

```diff
@@ -0,0 +1,600 @@
+            "item_id=%s error_message:%s",
+            item_id,
+            str(e))
+        raise UpdateWorkflowsForOraBlockException(str(e)) from e
+
+
+@log_task_info
+def update_workflow_for_submission(submission_uuid, assessment_requirements=None, course_settings=None):
+    """
+    Updates ORA workflow created for a given submission
+    Wrapper for `workflow.api.update_from_assessments(submission_uuid, assessment_requirements, course_override)`
+    """
+    try:
+
+        if assessment_requirements is None or course_settings is None:
+            peer_workflows = get_blocked_peer_workflows(submission_uuid)
+            if peer_workflows is not None:
+                workflow_update_data = get_workflow_update_data(peer_workflows)
+                submission_data = _get_submission_data(workflow_update_data, submission_uuid=submission_uuid)
+
+                # only one course with one ORA item expected for a given submission
+                course_settings = submission_data["course_settings"]
+                assessment_requirements = submission_data["assessment_requirements"]
+
+        api.update_from_assessments(submission_uuid, assessment_requirements, course_settings)
+
+        return WorkflowUpdateResult(message="ORA workflow update for a single blocked submission "
+                                            "completed successfully. ",
+                                    submission_uuid=submission_uuid)
+    except Exception as e:  # pylint: disable=broad-except
+        logger.error(
+            "ORA workflow update for a single submission failed. "
+            "submission_uuid=%s assessment_requirements=%s course_settings=%s. error_message=%s",
+            submission_uuid,
+            assessment_requirements,
+            course_settings,
+            str(e))
+        raise UpdateWorkflowForSubmissionException(str(e)) from e
+
+
+def is_flexible_peer_grading_on(ora_block, course_block):
+    """
+    Verify on ORA and Course level if flexible peer grading is "ON"
+
+    Args:
+        ora_block (OpenAssessmentBlock): ORA block
+        course_block (CourseBlock): Course block
+
+    Returns:
+        bool: True if given ORA is configured with flexible peer grading set "ON"
+    """
+    if ora_block is not None:
+        workflow_requirements = ora_block.workflow_requirements()
+        if workflow_requirements.get('peer') and workflow_requirements['peer'].get('enable_flexible_grading'):
+            return True
+    if course_block is not None:
+        return course_block.force_on_flexible_peer_openassessments
+
+    return False
+
+
+def get_blocked_peer_workflows(course_id=None, item_id=None, submission_uuid=None):
+    """
+    Retrieve ORA peer workflows not completed for >7 days
+
+    Returns:
+        list (PeerWorkflow): list of workfows not completed for > 7 days
+    """
+
+    filters = {
+        'created_at__lte': timezone.now() - datetime.timedelta(days=7),
+        'completed_at__isnull': True,
```

Comment:
Good thing I checked again. 
completed_at is the timestamp of when the learner completed the peer grades they needed to complete
grading_completed_at is the timestamp of when they recieved enough peer grades.
So, we want someone who HAS completed their assigned peer assessments and HASN'T recieved enough peer assessments.
```suggestion
        'grading_completed_at__isnull': True,
        'completed_at__isnull': False,
```